### DISCUSSION TITLE
1-bit quantization?

### DISCUSSION THREAD

**Discussion by PABannier** [ggml-org/ggml#754] [General]
A team from Microsoft recently came up with [1-bit quantization](https://huggingface.co/papers/2402.17764) drastically reducing memory footprint and token throughput. The weights are effectively encoded as a ternary bit {-1, 0, 1}. Reported results are super encouraging.

@ggerganov is it of interest for GGML? I can implement it and perform a bunch of benchmarks.

**Comment by Green-Sky** [comment]
https://github.com/ggerganov/llama.cpp/issues/5761 more on this topic :)
